src/gluonts/shell/sagemaker/params.py

```diff
@@ -79,3 +79,105 @@ def encode_sagemaker_parameters(decoded_params: dict) -> dict:
     {'foo': '[1, 2, 3]', 'bar': 'hello'}
     """
     return map_dct_values(encode_sagemaker_parameter, decoded_params)
+
+
+def detrim_and_decode_sagemaker_parameters(trimmed_params: dict) -> dict:
+    """Decode a SageMaker parameters dictionary where all values are strings.
+
+    Example:
+
+    >>> detrim_sagemaker_parameters({
+    ...     '_0_foo': '[1, ',
+    ...     '_1_foo': '2, 3',
+    ...     '_2_foo': ']',
+    ...     '_0_bar': 'hell',
+    ...     '_1_bar': 'o'
+    ... })
+    {'foo': [1, 2, 3], 'bar': 'hello'}
+    """
+    encoded_params = detrim_sagemaker_parameters(trimmed_params)
+    return map_dct_values(decode_sagemaker_parameter, encoded_params)
+
+
+def encode_and_trim_sagemaker_parameters(
+    decoded_params: dict, max_len: int = 256
+) -> dict:
+    """Encode a SageMaker parameters dictionary where all values are strings then
+    trim them to account for Sagemaker character size limit.
+
+    >>> encode_and_trim_sagemaker_parameters({
+    ...     "foo": [1, 2, 3],
+    ...     "bar": "hello"
+    ... }, max_len = 4)
+    {
+        '_0_foo': '[1, ',
+        '_1_foo': '2, 3',
+        '_2_foo': ']',
+        '_0_bar': 'hell',
+        '_1_bar': 'o'
```

Comment:
This is breaking doctest. Should we remove readability or I think we could add a doctest directive `NORMALIZE_WHITESPACE` ? Not familiar with doctest at all.

Reply:
I see, didn't know of that.

I think readability is more important here. If `NORMALIZE_WHITESPACE` works, I'm all for it.